python/ray/data/datasource/torch_datasource.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import TYPE_CHECKING, Callable, Iterator, List
+
+from ray.data.block import Block, BlockMetadata, T
+from ray.data.datasource import Datasource, ReadTask
+
+if TYPE_CHECKING:
+    import torch.utils.data
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleTorchDatasource(Datasource[T]):
+    """A datasource that let's you use Torch datasets with Ray Data.
+
+    .. warning::
+        ``SimpleTorchDatasource`` doesn't support parallel reads. You should only use
+        this datasource for small datasets like MNIST or CIFAR.
+
+    Example:
+        >>> import ray
+        >>> from ray.data.datasource import SimpleTorchDatasource
+        >>>
+        >>> dataset_factory = lambda: torchvision.datasets.MNIST("data", download=True)
+        >>> dataset = ray.data.read_datasource(
+        ...     SimpleTorchDatasource(), parallelism=1, dataset_factory=dataset_factory
+        ... )
+        >>> dataset.take(1)
```

Comment:
Looks like this example is failing in CI, could you add `# doctest: +SKIP` to all but the import lines? E.g. https://github.com/ray-project/ray/blob/cadbd1fb4e2a19e5e8bbd0e9bca9c673f8d0856c/python/ray/data/datasource/parquet_datasource.py#L168-L173

Reply:
Fixed